allensdk/brain_observatory/behavior/data_objects/eye_tracking/rig_geometry.py

```diff
@@ -0,0 +1,220 @@
+import warnings
+
+import numpy as np
+import pandas as pd
+from typing import List, Optional
+
+import pynwb
+from pynwb import NWBFile
+
+from allensdk.brain_observatory.behavior.data_objects import DataObject
+from allensdk.brain_observatory.behavior.data_objects.base \
+    .readable_interfaces import \
+    LimsReadableInterface, JsonReadableInterface, NwbReadableInterface
+from allensdk.brain_observatory.behavior.data_objects.base\
+    .writable_interfaces import \
+    NwbWritableInterface
+from allensdk.brain_observatory.behavior.schemas import \
+    OphysEyeTrackingRigMetadataSchema
+from allensdk.brain_observatory.nwb import load_pynwb_extension
+from allensdk.internal.api import PostgresQueryMixin
+
+
+class RigGeometry(DataObject, LimsReadableInterface, JsonReadableInterface,
+                  NwbReadableInterface, NwbWritableInterface):
+    def __init__(self, equipment: str,
+                 monitor_position_mm: List,
+                 monitor_rotation_deg: List,
+                 camera_position_mm: List,
+                 camera_rotation_deg: List,
+                 led_position: List):
+        super().__init__(name='rig_geometry', value=self)
+        self._monitor_position_mm = monitor_position_mm
+        self._monitor_rotation_deg = monitor_rotation_deg
+        self._camera_position_mm = camera_position_mm
+        self._camera_rotation_deg = camera_rotation_deg
+        self._led_position = led_position
+        self._equipment = equipment
```

Comment:
These values are not actually used anywhere that I know of where the length matters, but yes they are expected to be the same shape. Writing to NWB would fail if not the correct length; I guess it wouldn't hurt to add validation for that here.

Reply:
I added a `Coordinates` type which enforces that all coordinates contain x,y,z